climt/_core/constants.py

```diff
@@ -0,0 +1,303 @@
+
+    miscellaneous_constants=set(())
+)
+
+
+class ConstantLibrary(object):
+    """
+    Maintain constant values and categories. Currently available
+    constants are accessible as :code:`constant_library.current_constants`.
+    """
+
+    def __init__(self):
+
+        self.init_constant_library()
+
+    def init_constant_library(self):
+        """
+        Initialise the constant library from sympl's :code:`default_constants`.
+        """
+
+        self.current_constants = deepcopy(default_constants)
+
+        self.constant_categories = deepcopy(default_categories)
+
+        for climt_name in constant_name_map.keys():
+            self.current_constants[climt_name] = deepcopy(
+                default_constants[constant_name_map[climt_name]])
+
+    def reset_constant_library(self):
+        """
+        Remove all additions to constant library that have happened
+        during runtime.
+        """
+
+        self.init_constant_library()
+
+    def _list_constants(self, constant_list):
+
+        for quantity in constant_list:
+            print("{}: {} {}".format(quantity,
+                                     self.current_constants[quantity].values,
+                                     self.current_constants[quantity].units))
+            print()
+
+    def list_available_constants(self):
+        """
+        List all constants currently available in the library, along
+        with the category to which they belong.
+
+        """
+
+        print('Physical Constants')
+        print('===========================')
+        self._list_constants(self.constant_categories['physical_constants'])
+        print()
+        print('Chemical Constants')
+        print('===========================')
+        self._list_constants(self.constant_categories["chemical_constants"])
+        print()
+        print('Planetary Constants')
+        print('===========================')
+        self._list_constants(self.constant_categories["planetary_constants"])
+        print()
+        print('Stellar Constants')
+        print('===========================')
+        self._list_constants(self.constant_categories["stellar_constants"])
+        print()
+        print('Atmospheric Constants')
+        print('===========================')
+        self._list_constants(self.constant_categories["atmospheric_constants"])
+        print()
+        print('Condensible Constants')
+        print('===========================')
+        self._list_constants(self.constant_categories["condensible_constants"])
+        print()
+        print('Oceanographic Constants')
+        print('===========================')
+        self._list_constants(self.constant_categories["oceanographic_constants"])
+        print()
+        print('Miscellaneous Constants')
+        print('===========================')
+        self._list_constants(self.constant_categories["miscellaneous_constants"])
+        print()
+
+    def add_constants_to_library(self, constant_descriptions):
```

Comment:
What is the use case for this method? Should it be modifying Sympl's constants dictionary?